statsmodels/formula/tests/test_formula.py

```diff
@@ -225,3 +225,16 @@ def test_predict_nondataframe():
     error = patsy.PatsyError
     with pytest.raises(error):
         fit.predict([0.25])
+
+
+def test_formula_environment():
+    df = pd.DataFrame({'x': [1, 2, 3], 'y': [2, 4, 6]})
+    env = patsy.EvalEnvironment({})
+
+    try:
```

Comment:
You can remove the try except here and just have the OLS. 

You could fit the model and check that 'x' is a parameter name.